src/algorithms/tonal/audio2pitch.cpp

```diff
@@ -0,0 +1,130 @@
+#include "audio2pitch.h"
+
+using namespace essentia;
+using namespace standard;
+
+const char* Audio2Pitch::name = "Audio2Pitch";
+const char* Audio2Pitch::category = "Pitch";
+const char* Audio2Pitch::description = DOC("Extractor algorithm to compute pitch with several possible pitch algorithms, specifically targeted for real-time pitch detection on saxophone signals.");
+
+bool Audio2Pitch::isAboveThresholds(Real pitchConfidence, Real loudness) {
+  return (pitchConfidence >= _pitchConfidenceThreshold) && (loudness >= _loudnessThreshold);
+}
+
+void Audio2Pitch::configure() {
+
+  _sampleRate = parameter("sampleRate").toReal();
+  _frameSize = parameter("frameSize").toInt();
+  _minFrequency = parameter("minFrequency").toReal();
+  _maxFrequency = parameter("maxFrequency").toReal();
+  _pitchAlgorithmName = parameter("pitchAlgorithm").toString();
+  _loudnessAlgorithmName = parameter("loudnessAlgorithm").toString();
+  _tolerance = parameter("tolerance").toReal();
+  _pitchConfidenceThreshold = parameter("pitchConfidenceThreshold").toReal();
+  _loudnessThreshold = parameter("loudnessThreshold").toReal();
+
+  if (_maxFrequency > _sampleRate*0.5 ) {
+    throw EssentiaException("Audio2Pitch: Max frequency cannot be higher than Nyquist frequency");
+  }
+  if (_maxFrequency <= _minFrequency) {
+    throw EssentiaException("Audio2Pitch: Max frequency cannot be lower than min frequency");
+  }
+
+  if (_pitchAlgorithmName != "pyin_fft" && _pitchAlgorithmName != "pyin") {
+    E_INFO("Audio2Pitch: 'pitchAlgorithm' = "<<_pitchAlgorithmName<<"\n");
+    throw EssentiaException("Audio2Pitch: Bad 'pitchAlgorithm' parameter");
+  }
+  if (_pitchAlgorithmName == "pyin_fft") _isSpectral = true;
+  if (_pitchAlgorithmName == "pyin") _isSpectral = false;
+
+  if (_isSpectral) {
+    _windowing = AlgorithmFactory::create("Windowing");
+    _spectrum = AlgorithmFactory::create("Spectrum");
+    _pitchAlgorithm = AlgorithmFactory::create("PitchYinFFT");
+
+    _windowing->configure("type", "hann",
+                          "size", _frameSize);
+    _spectrum->configure("size", _frameSize);
+  }
+  else {
+    _pitchAlgorithm = AlgorithmFactory::create("PitchYin");
+  }
+
+  if (_loudnessAlgorithmName == "loudness") {
+    _loudnessAlgorithm = AlgorithmFactory::create("Loudness");
+  }
+  else if (_loudnessAlgorithmName == "rms") {
+    _loudnessAlgorithm = AlgorithmFactory::create("RMS");
+  }
+  else {
+    E_INFO("Audio2Pitch: 'loudnessAlgorithm' = "<<_loudnessAlgorithmName<<"\n");
+    throw EssentiaException("Audio2Pitch: Bad 'loudnessAlgorithm' parameter");
+  }
+
+  // switch between pyin and pyin_fft to propagate the weighting parameter
```

Comment:
Here we are directly using parameter strings for comparison, but `isSpectral` is used in other parts of the code. Is this because there could be multiple FFT-based algorithms in the future? If not, we could simplify and get rid of `isSpectral`.

Reply:
well, this was because we started including other algorithms like PitchMelodia and in the python prototype we have more algorithms based on spectral processing. I think it might be removed if we don't plan to include others.

Reply:
Let's get rid of `isSpectral` to simplify.